app/src/main/java/org/mozilla/fenix/home/sessioncontrol/SessionControlAdapter.kt

```diff
@@ -134,7 +136,8 @@ sealed class AdapterItem(@LayoutRes val viewType: Int) {
         val isLastTab: Boolean
     ) : AdapterItem(TabInCollectionViewHolder.LAYOUT_ID) {
         override fun sameAs(other: AdapterItem) =
-            other is TabInCollectionItem && tab.id == other.tab.id
+            other is TabInCollectionItem && tab.id == other.tab.id &&
+                this.collection == other.collection
```

Comment:
@Alexandru2909 After https://github.com/mozilla-mobile/fenix/pull/25229 adding a new check for this item it seem like we don't need this change anymore?

Just deleting some tabs from a collection and then the collection
Followed by restoring the collection and checking the restored tabs also
Seem to work okay for me with just the changes from the collection item.

Reply:
I can confirm the change is not necessary anymore. I removed it and updated the PR. Thank you !